Mage.Sets/src/mage/cards/c/CuratorsWard.java

```diff
@@ -0,0 +1,120 @@
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.c;
+
+import java.util.UUID;
+import mage.constants.SubType;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.constants.Outcome;
+import mage.target.TargetPermanent;
+import mage.abilities.keyword.EnchantAbility;
+import mage.abilities.keyword.HexproofAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AttachmentType;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author keelahnkhan
+ */
+public class CuratorsWard extends CardImpl {
+
+    public CuratorsWard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{U}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant permanent
+        TargetPermanent auraTarget = new TargetPermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.AddAbility));
+        Ability ability = new EnchantAbility(auraTarget.getTargetName());
+        this.addAbility(ability);
+
+        // Enchanted permanent has hexproof.
+        ability.addEffect(new GainAbilityAttachedEffect(HexproofAbility.getInstance(), AttachmentType.AURA).setText("Enchanted permanent has hexproof"));
+        
+        // When enchanted permanent leaves the battlefield, if it was historic, draw two cards.
+        this.addAbility(new CuratorsWardTriggeredAbility());
+    }
+
+    public CuratorsWard(final CuratorsWard card) {
+        super(card);
+    }
+
+    @Override
+    public CuratorsWard copy() {
+        return new CuratorsWard(this);
+    }
+}
+
+class CuratorsWardTriggeredAbility extends TriggeredAbilityImpl {
+
+    public CuratorsWardTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(2).setText("When enchanted permanent leaves the battlefield, if it was historic, draw two cards."), false);
```

Comment:
You don't need to do `setText`, instead add `staticText = "When enchanted permanent leaves the battlefield, if it was historic, draw two cards."` to the constructor.

Reply:
I tried but it doesn't seem like staticText is an inherited variable of TriggeredAbilityImpl? I couldn't pass the text to the DrawCardSourceControllerEffect constructor either. I might be being oblivious here.

Reply:
You're right, instead add
```@Override public String getRule(){return "When enchanted permanent leaves the battlefield, if it was historic, draw two cards.";}

Reply:
Gotcha, will do!